src/main/java/dev/morling/onebrc/CalculateAverage_ericxiao.java

```diff
@@ -125,31 +94,48 @@ public void add(long keyStart, long keyEnd, long valueEnd) {
             }
             int value = multiplier * accumulator;
 
-            hashMap.compute(key, (k, v) -> {
-                if (v == null) {
-                    k.materialize();
-                    return new int[]{ value, value, value, 1 };
-                }
-                else {
-                    v[0] = Math.min(v[0], value);
-                    v[1] = Math.max(v[1], value);
-                    v[2] += value;
-                    v[3]++;
-                    return v;
-                }
-            });
+            // Calculate station
+            long hash = 0;
+            for (int i = 0; i < keyLength; i++) {
+                hash = 31 * hash + (int) entryBytes[i];
+            }
+            // if (hash < 0) {
+            // System.out.println("stop");
+            // }
+
+            int stationHash = (int) (Math.abs(hash) % 10000);
```

Comment:
use stations.length

also could we try this 

https://lemire.me/blog/2016/06/27/a-fast-alternative-to-the-modulo-reduction/

Reply:
see https://stackoverflow.com/questions/6670715/mod-of-power-2-on-bitwise-operators

Reply:
> actually we can probably do even better if we ensure the table length is a power of 2.

Oh ya good point.